### PR TITLE
kernel: make MULTITHREADING promptless if single-thread not supported

### PR DESCRIPTION
**Pull request by ioannisg**
If single thread builds are not supported by the
architecture, the MULTITHREADING options should be
prompt-less, to avoid modifications. We also introduce
an explicit ARCH-level Kconfig that reflects whether
the ARCH is capable of single-thread Zephyr builds.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
This will block enabling single-threaded builds on ARCHes that do not support it. If other arches besides Cortex-M support single-threaded mode, the maintainers need to explicitly select the capability option in Kconfig.

**Review by nvlsianpu — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

File: arch/Kconfig [hunk 1]
```diff
@@ -464,6 +464,9 @@ endmenu
 # Architecture Capabilities
 #
 
+config ARCH_HAS_NO_MULTITHREADING_SUPPORT
```
Comment (nashif) [inline arch/Kconfig:467]: this config sounds and reads wrong. I read it as: `This architecture does not support multi-threading.`

Maybe it should be 'ARCH_HAS_SINGLE_THREAD_SUPPORT' or something along those lines.

**Review by nashif — CHANGES_REQUESTED** [review]

File: arch/Kconfig (hunk 1 above)
Comment (ioannisg) [reply arch/Kconfig:467]: Ah, that's right, @nashif , it's funny, actually :)
Yeap, your suggestion is fine.

Comment (ioannisg) [reply arch/Kconfig]: Done @nashif @galak

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**